letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -458,7 +471,10 @@ def _create_vhost(self, path):
                 is_ssl = True
 
         filename = get_file_path(path)
-        is_enabled = self.is_site_enabled(filename)
+        if self.conf("handle-sites"):
+            is_enabled = self.is_site_enabled(filename)
+        else:
+            is_enabled = True
```

Comment:
The described scenario is possible, and it will also enable users to request certificates for sites that aren't currently enabled (temporarily disabled for maintenance reasons or whatnots), but exist in the sites-available.

My view of the semantics differs from yours; the site enabling concept is not in vanilla apache, and even though the apache plugin was originally created towards Debian based distributions I think they are the kind of corner case here. Hence semantically it wouldn't make sense to call a method regarding enabling in the context that doesn't know anything about enabling.